project/src/main/webapp/event-details.js

```diff
@@ -1,32 +1,47 @@
-const eventHost = 'test@example.com'; // hard-coded event host value
-let loginState = null;
-let currentUser = null;
-
-window.onload = function onLoad() {
+window.onload = async function onLoad() {
   getEventDetails();
-  checkLoginStatus();
-  getVolunteeringOpportunities();
-  populateOpportunitiesDropdown();
+  const eventHost = await getEventHost();
+  const loginStatus = await getLoginStatus();
+  populateVolunteeringOpportunitiesUI(eventHost, loginStatus);
+  showCreateOpportunityLink(eventHost, loginStatus);
+  setSignupAction();
 };
 
 
 /**
- * Check the current login status to update
- * isLoggedIn and currentUser.
+ * Returns the current login status.
  */
-async function checkLoginStatus() {
+async function getLoginStatus() {
```

Comment:
There should be one of these in shared.js

Reply:
Done -  added another function called getCurrentUserEmail()